JoomGalleryfriends/JG4-dev
# Dynamic processing of images in the frontend

This PR adds the functionality of dynamically process images before showing them in the frontend.

What you have to do to activate this functionality:
1. The parameter `Use real paths` in the joomgallery configuration has to be set to `No` (General settings/Common settings)
2. In `Image processing (dynamic images)` under General settings/Image processing you have to add the imagetype you want to dynamically process and adjust the settings there.
3. In case of watermarking: Additionally `Add Watermark` in the parameters section of the corresponding category in the category edit form

Can you think of any other setting you want to allow/disallow in the category and/or image parameter section?

## How to test this PR
- Activate the dynamic processing functionality as described above.
- Go to the frontend and visit a view showing an imagetype which needs to be dynamically processed.
- Check if the processing was correctly done.

### Example:
1. Upload some images to the Uncategorized category (make sure they dont get a static watermark during uploading)
2. Activate dynamic processing for detail images in the configs
![grafik](https://github.com/JoomGalleryfriends/JG4-dev/assets/39154009/159e4123-f640-42a8-b688-0468e462aa3d)
3. Add a watermark to the detail images in the configs
![grafik](https://github.com/JoomGalleryfriends/JG4-dev/assets/39154009/57825782-3515-46b3-ae74-9eab5b679599)
4. Set Add Watermark to yes in the params section of the Uncategorized category
![grafik](https://github.com/JoomGalleryfriends/JG4-dev/assets/39154009/2fb59f78-4eb5-49ba-b90c-7c1321b684ab)

A watermark is now dynamically applied to all the detail images of the Uncategorized category when accessing in the frontend.
